### DISCUSSION TITLE
Migrations with a different identifier

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [directus/directus#9255] [Feature Requests]
I was writing some custom migrations and I noticed that if I want to add a migration and then update the Directus version, I need to go to the repo and check the last day to make sure to migration runs before the next one Directus custom migrations. 

What do you think of using migration by tags instead of date, in that way I could just add migration to my current version as `88-migration-information.ts` and the custom migration could be `88-custom-migration-information.ts`

I know it is not a big ideal but was just a thought.

**Comment by rijkvanzanten** [comment]
Wouldn't that have the same problem? If we'd call it "99-add-webhooks-headers", you'd still have the ordering problem 🤔

**Reply by juancarlosjr97** [reply]
My thought is that if the custom migrations are using numbers instead of dates, it would be easier to create custom migrations without checking the latest migration to avoid crashing.

If I am doing a custom migration using v88, I would be creating it as `88B-xyz.js` because `89` will have their own migrations

**Reply by rijkvanzanten** [reply]
Ahhh, you could already do that! you can make the identifier whatever you want 🙂

**Reply by juancarlosjr97** [reply]
Is not the identifier name structure as `date-description.js` or are you referring to something differently?

**Reply by rijkvanzanten** [reply]
> Is not the identifier name structure as `date-description.js` or are you referring to something differently?

The structure is `{identifier}-{description}`, but there's no rules around what the identifier is supposed to be, we just happen to use creation date to make sure it runs in order 👍🏻

**Reply by juancarlosjr97** [reply]
Got it, but if a migration is created how to make sure is executed before the migrations of the next version??? 

Lets assume today I am creating a new migration, then run fine, I update the version and there is one migration from today too in the next version, that could cause a crash if it is applied after all the migration of the system.

**Reply by rijkvanzanten** [reply]
Hmm good one, you'd have to keep track of the existing system migrations and make sure your custom identifier sits in the middle. For what it's worth, we won't be creating any more migrations with backdated IDs. That was a one-time screw-up in the migrations for the insights module 👍🏻

**Reply by juancarlosjr97** [reply]
Got it, thank you @rijkvanzanten!!!

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
